blockstore/src/main/java/brooklyn/location/blockstore/azure/arm/AzureArmVolumeManager.java

```diff
@@ -0,0 +1,411 @@
+        return vmApi.get(vmName) != null;
+    }
+
+    private boolean waitForJobToComplete(final AzureComputeApi api, final URI uri, Duration waitTime) {
+        checkNotNull(uri, "uri must not be null");
+        final JobApi jobApi = api.getJobApi();
+        final AtomicReference<JobStatus> latest = new AtomicReference<JobStatus>();
+        
+        boolean done = Repeater.create("Waiting job for URI to completee: " + uri)
+                .every(1, TimeUnit.SECONDS)
+                .limitTimeTo(waitTime)
+                .until(new Callable<Boolean>() {
+                    @Override
+                    public Boolean call() throws Exception {
+                        JobStatus currentJobStatus = jobApi.jobStatus(uri);
+                        latest.set(currentJobStatus);
+                        return currentJobStatus == ParseJobStatus.JobStatus.DONE;
+                    }
+                })
+                .run();
+        return done;
+    }
+
+    private static class AzureArmBlockDevice implements BlockDevice {
+
+        private static final Logger LOG = LoggerFactory.getLogger(AzureArmBlockDevice.class);
+
+        private final JcloudsLocation location;
+        private final DataDisk disk;
+        private final String resourceGroupName;
+        private final String storageAccountName;
+        
+        private AzureArmBlockDevice(JcloudsLocation location, DataDisk disk, String resourceGroupName, String storageAccountName) {
+            this.location = checkNotNull(location, "location");
+            this.disk = checkNotNull(disk, "disk");
+            this.resourceGroupName = checkNotNull(resourceGroupName, "resourceGroupName");
+            this.storageAccountName = checkNotNull(storageAccountName, "storageAccountName");
+        }
+
+        @Override
+        public String getId() {
+            return disk.name();
+        }
+
+        @Override
+        public JcloudsLocation getLocation() {
+            return location;
+        }
+
+        public DataDisk getDisk() {
+            return disk;
+        }
+
+        public String getResourceGroupName() {
+            return resourceGroupName;
+        }
+        
+        public String getStorageAccountName() {
+            return storageAccountName;
+        }
+        
+        @Override
+        public AzureArmAttachedBlockDevice attachedTo(JcloudsMachineLocation machine, String deviceName) {
+            if (!machine.getParent().equals(location)) {
+                LOG.warn("Attaching device to machine in different location to its creation: id={}, location={}, machine={}",
+                        new Object[]{getId(), location, machine});
+            }
+            return new AzureArmAttachedBlockDevice(machine, disk, resourceGroupName, storageAccountName, deviceName);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
```

Comment:
I hate Guava for this :D
Shouldn't we avoid to depend on those syntactic sugar coming from Guava when not strictly needed.
Of course, no strong feelings, feel free to ignore me